dev/io.openliberty.org.jboss.weld5/build.gradle

```diff
@@ -0,0 +1,161 @@
+
+def unzippDir = "${buildDir}/unzip"
+def pathToOSManifest = "${unzippDir}/META-INF/MANIFEST.MF"
+def pathToTemplateManifest = "${buildDir}/../../bnd.template"
+def tempPathToNewManifest = "${buildDir}/check-me-in-if-i-change.bnd"
+def pathToNewManifest = "${buildDir}/../../check-me-in-if-i-change.bnd"
+
+
+/////// Modifications to the bnd begin here ///////////////////
+
+def additionalExports = """\
+ org.jboss.weldx.transaction;version=\\"${weldVersion}\\",\\\\
+ org.jboss.weld.bean.proxy.util;version=\\"${weldVersion}\\",\\\\
+ org.jboss.weld.annotated.enhanced;version=\\"${weldVersion}\\",\\\\
+ org.jboss.weld.executor;version=\\"${weldVersion}\\",\\\\
+ org.jboss.weld.metadata;version=\\"${weldVersion}\\",\\\\
+ org.jboss.weld.resolution;version=\\"${weldVersion}\\",\\\\
+ org.jboss.weld.resources;version=\\"${weldVersion}\\",\\\\
+ org.jboss.weld.util.collections;version=\\"${weldVersion}\\",\\\\
+"""
+
+def additionalImports = """\
+ javax.security.auth,\\\\
+ jakarta.xml.ws;version="[4.0,5)",\\\\
+"""
+
+// Key is a property, value is an addition.
+def additionalPropertiesForExports = [
+  "org.jboss.weld.event": "thread-context=true"
+]
+
+//////// Modifications end here ////////////////
+
+
+configurations {
+  weld {
+    transitive = false
+  }
+}
+
+dependencies {
+  weld "org.jboss.weld:weld-osgi-bundle:${weldVersion}"
+}
+
+task bndMagic {
+
+    doFirst{
+        copy {   
+            configurations.weld.asFileTree.each {
+               from(zipTree(it))
+            }
+            into unzippDir
+        }
+
+        def upstreamExports = extractFromManifest("Export-Package:")
+        def upstreamImports = extractFromManifest("Import-Package:")
+
+        //Make updates to existing exports (currently we have no updates for imports)
+        for (int i = 0; i < upstreamExports.size(); i++) {
+            for (String key : additionalPropertiesForExports.keySet()) {
+                String[] split = upstreamExports.get(i).split(";")
+                if (split[0].equals(key)) {                  
+                    split[0] = split[0] + ";" + additionalPropertiesForExports.get(key)
+                    String updatedEntry = String.join(";", split)
+                    upstreamExports.set(i, updatedEntry)
+                }
+            }
+        }
+
+        // Compile the full exports string. Consisting of: 
+        // 1) The new exports we add - a trailing new line char is included in the string
+        // A single whitespace charachter because String.join won't do a prefix to the first entry.
+        // Every entry followed by a ',\' a line break, and a whitespace to start the next line.
+        String exportsBndFormat = additionalExports + " " + String.join(",\\\\" + System.lineSeparator() + " ", upstreamExports)
+
+        // The input string uses the same format
+        String importsBndFormat = additionalImports + " " + String.join(",\\\\" + System.lineSeparator() + " ", upstreamImports)
```

Comment:
I don't _think_ we want `System.lineSeparator()` here. We don't want the bnd file to have `\r\n` if it was generated on Windows.